Modified instructions for the modified submission form.

hepcat72 opened the pull request:
## Summary Change Description

The submission form was edited in google docs to request LCMethod details and the documented instructions were changed to reflect that.

## Affected Issues/Pull Requests

- Resolves #705

## Review Notes

Note, I edited the submission form.  Please take a look at that.
- I edited the [link in section 1](http://tracebase.princeton.edu/DataRepo/upload) from `tracebase-dev` to `tracebase`
- I changed the LC Method(s) question to read:
  > What liquid chromatography method and run length did you use for each file (e.g. polar-HILIC-25-min, polar-reversed-phase-ion-pairing-25-min, polar-reversed-phase-25-min, lipid-reversed-phase-25-min)?
If none of these, please include a description (e.g. lipid-reversed-phase-25-min This method uses C18 columns for the analysis of lipids).
  I tried to indicate that the user should use the supplied example values, with run-length, and that if their method differs, they should also include a description.
- I changed the answer from short answer to long answer (paragraph)

Personally, I feel like this data should be in a sheet in the uploaded excel file so that each method can be associated with the appropriate sample/file, but I tried to stay true to the issue description.

## Checklist
<!--
If any of the checkbox requirements are not met, uncheck them and add an
explanation. E.g. Linting errors pre-date this PR.
-->
This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 <!-- Edit as desired (e.g. based on complexity) -->
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    <!--
    Require reviewers with specific expertise to be included among the minimum
    number of reviewers by tagging them.  Also please send them a message.
    -->
  - Review period: 2 days <!-- Edit as desired (e.g. based on complexity) -->
- Associated issue/pull request requirements:
  <!--
  Assert that all requirements in issues marked "resolved" are done and that
  all required pull requests are merged.  If any are not done, either edit or
  split the issue or explain the unmerged affected pull requests.
  -->
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  <!--
  Uncheck items to acknowledge failures/conflicts you intend to address.
  Add an explanation if any won't be addressed before merge.
  -->
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  <!--
  These are additional requirements that are not directly associated with
  resolving the affected issue(s).
  -->
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [x] Added changes to *Unreleased* section of [`changelog.md`](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

hepcat72 commented on TraceBaseDocs/TraceBaseDocs/Upload/How to Upload.md:
```diff
@@ -43,8 +43,8 @@ the TraceBase Animal and Sample Table Template.
    2. Link to Google Drive folder with study submission.
-   3. Mass Spectrometry Details (see [How to label Mass Spec Run
-      Information](https://docs.google.com/document/d/1Lm4br-jCB2QwbgPyzJvvgLgaO-t7Cwcq1xHyBIOlTog/edit?usp=sharing))
+   3. Mass Spectrometry Details (see [[How to label Mass Spec Run
+      Information]])
```
C. Removed the google docs link and added a new markdown file to paste its contents in.  I have not made any changes to the documentation website.

hepcat72 replied:
I assume that this internal "link" will work.  Since I have not deployed this to the docs website, I don't know how to test that it works.

hepcat72 commented on TraceBaseDocs/TraceBaseDocs/Upload/How to label Mass Spec Run Information.md:
C. This content was copied from the [google doc](https://docs.google.com/document/d/1Lm4br-jCB2QwbgPyzJvvgLgaO-t7Cwcq1xHyBIOlTog/edit?usp=sharing) and then edited.

hepcat72 commented on TraceBaseDocs/TraceBaseDocs/Upload/How to label Mass Spec Run Information.md:
```diff
@@ -0,0 +1,35 @@
+
+## LC Method(s)
+
+- Description of Liquid Chromatography method and run length used
+- Possible values:
+  - unknown
+  - polar-HILIC-25-min
+  - polar-reversed-phase-ion-pairing-25-min
+  - polar-reversed-phase-25-min
+  - lipid-reversed-phase-25-min
+- If the method/run-length is not available, please also include a description
+  - Example Description: lipid-reversed-phase-25-min
+    This method involves the analysis of lipids using reversed-phase
+    chromatography on C18 columns, coupled with high resolution mass
+    spectrometry. It is mainly used for the analysis of lipids.
```
C. This is the section that was edited

hepcat72 commented on TraceBaseDocs/TraceBaseDocs/Upload/How to label Mass Spec Run Information.md:
```diff
@@ -0,0 +1,35 @@
+# How to label Mass Spec Run Information
+
```
C. Added a title.

lparsons reviewed (commented):
Thanks for catching and fixing the link to tracebase-dev. Also, I agree that this should probably be a new sheet added to the template. The issue #705  states "Update sumbission templates..." so I don't see any conflict in doing that, but please let me know if you have questions or concerns.

hepcat72 commented:
> I agree that this should probably be a new sheet added to the template. The issue #705 states "Update sumbission templates..." so I don't see any conflict in doing that, but please let me know if you have questions or concerns.

The ambiguous/confusing part to me about this is, I wasn't sure it the animal/sample template was included in "submission templates" or if the qualifier "submission" was singling out the sheet associated with the submission form by name and that it was plural because it was "allowing" a separate submission template to be added (separate from the animal/sample sheet) for the newly required information, given the M:M association between the sample and LCMethod / MSRun.  I can see you could make a case that that is over-complicating the issue, but given I was struggling with reconciling this question with other issues/concerns, it just wasn't clear to me.  So thanks for providing your interpretation.  It sounds like you're suggesting that the design here is given a wide berth of interpretations, so I will come up with what I think is the most appropriate solution and possibly revise this and request a followup review.

lparsons commented:
Thanks Rob. I suggest that run your design past the group before committing significant time to it. As we've reviewed before, providing a means for researchers to provide a single lcmethod for each accucor file should cover the majority of cases. When we write the loading code, we can allow for finer grained control (see #706). However, it is best to keep things as simple and straightforward for researchers.

hepcat72 commented:
> Thanks Rob. I suggest that run your design past the group before committing significant time to it. As we've reviewed before, providing a means for researchers to provide a single lcmethod for each accucor file should cover the majority of cases. When we write the loading code, we can allow for finer grained control (see #706). However, it is best to keep things as simple and straightforward for researchers.

I am a step ahead of you on that. I created a design proposal yesterday before I left, for issue #706.  There is a lot of overlap between this issue and that one, if you implement the changes called for here in the animal/sample table, since it's the loading code of the accucor data loader that takes in that data.  And that's often how I approach design for challenges like this: from the perspective of the code.  It informs what I need in terms of the data structure required.  Since I was in the mindset of the submission form with this issue, and it didn't seem to fit with the data relationship between LC Method and the submission form, I was having trouble reconciling what seemed like contradictory requirements.  Looking at it from the perspective of the loading code helped clarify it for me.

The design I created in issue #706 relates to the changes we discussed yesterday in #687, where I liked the suggestion you made to add an optional `sample data header` column.  That design proposal suggests adding that value as a "name" field to an `MSRun` record (or equivalent ms-related table - I have to review the schema design to get familiar with the model names you assigned).  What that design doesn't explicitly say is that instead of adding that `sample data header` column to the animal/sample table, I instead propose supplying a separate LCMS metadata file to the accucor loader that associates sample names to `sample data header`s, along with the other metadata delineated in the issue description (AccuCor file, mzXML filename, Researcher, Date, Instrument, lc method).

This is my proposed alternative to re-supplying the animal/sample table to the accucor data loader.  And my reason for that relates to a [comment I made](https://github.com/Princeton-LSI-ResearchComputing/tracebase/issues/684#issuecomment-1737701278) that points out that additional data submissions, if we re-supply the animal/sample table, would require either modifying the original sample table associated with a previous submission or duplicate the samples in new submission, neither option of which seemed desirable to me.  This alternative design, adding an LCMS metadata file separates the dependency, keeps the animal/sample table clean and uncluttered, decouples the animal/sample table loading code from the accucor code, and meets the data relationship problems without duplicating data or modifying old data.

hepcat72 commented:
@lparsons - The dependencies have been merged to main and I updated this branch. Do you want to review it? Your last review was a comment, so I'll wait for the approval before merging.

lparsons reviewed (approved):
This looks good to me, thanks for moving the docs from Google Docs to the markdown files. I'm not sure if you modified the form or not, but it seems like we should probably update the form to take a single value for date, operator, and lc_method and indicate that if multiple are used, the researcher should fill in the details in a separate file. Am I correct in assuming that is how you were planning to have people submit more complicated situations? If so, let's go ahead and merge and then update the form and provide a template.

hepcat72 commented:
I guess the review was a tad premature.  I should have read my affected issues section.  I made some changes to the docs based on the changes in #774.  Take a look.

I also updated the [submission form](https://docs.google.com/forms/d/1Y2SEQgLTd8QUjbXGCuo3989WQxMfsfh4AbJyeL3f6c8/edit) and I added a template for the [LCMS-metadata file](https://docs.google.com/spreadsheets/d/1rfKOGqms8LPeqORO5gyTXLXDU2lvz-CG2aCEwmu8xHw/copy).

If you see anything there, let me know.

lparsons commented on TraceBaseDocs/TraceBaseDocs/Upload/Sample Information Sheet.md:
```diff
@@ -105,7 +105,8 @@ between their studies.
 ### Study Name
 
-An identifier for the “experiment” or collection of animals that this animal belongs to.
+An identifier for the “experiment” or collection of animals that this animal
```
```suggestion
An identifier for the "experiment" or collection of animals that this animal
```

hepcat72 replied:
Huh.  Looks like this pervades the docs:
```
$ git grep "“"
TraceBaseDocs/TraceBaseDocs/Overview/Navigating TraceBase.md:including “glucose” and tissues including “serum”.  This produces a result with
TraceBaseDocs/TraceBaseDocs/Overview/Navigating TraceBase.md:To get an overview of what is included, you can click the blue “Stats” button
TraceBaseDocs/TraceBaseDocs/Types of Data Output/Fcirc.md:* Note that for “intact” Fcirc values, this is different than “any
TraceBaseDocs/TraceBaseDocs/Types of Data Output/How to Download.md:All the data in TraceBase can be downloaded by clicking the “Download” button
TraceBaseDocs/TraceBaseDocs/Types of Data Output/How to Download.md:TraceBase metadata found in other pages (e.g. “Studies” table) can be
TraceBaseDocs/TraceBaseDocs/Upload/Labeling and Organizing Data.md:For a new experiment, create a new identifier counting up from “study001”, and
TraceBaseDocs/TraceBaseDocs/Upload/Labeling and Organizing Data.md:  “study001b_hyperthyroid lactate infusion”.
TraceBaseDocs/TraceBaseDocs/Upload/Labeling and Organizing Data.md:- Any text could be used in place of “study”.
TraceBaseDocs/TraceBaseDocs/Upload/Labeling and Organizing Data.md:Within each experiment, animal ID counts up from 001 “study001_m01, study001_m02…”.
TraceBaseDocs/TraceBaseDocs/Upload/Labeling and Organizing Data.md:  - E.g. “study001_m01_quad, study001_m01_tailserum_000,
TraceBaseDocs/TraceBaseDocs/Upload/Labeling and Organizing Data.md:into one folder labeled for each study “study001_my first infusion”,
TraceBaseDocs/TraceBaseDocs/Upload/Labeling and Organizing Data.md:“study001b_fixing my first infusion”.  Put everything related to the experiment
TraceBaseDocs/TraceBaseDocs/Upload/Labeling and Organizing Data.md:  minimal info “Q1, Q2”.
TraceBaseDocs/TraceBaseDocs/Upload/Sample Information Sheet.md:genotype as “unknown” (e.g. if the animal is a mixed background wildtype).
TraceBaseDocs/TraceBaseDocs/Upload/Sample Information Sheet.md:“infusate” separated by semicolons
TraceBaseDocs/TraceBaseDocs/Upload/Sample Information Sheet.md:* Example:  “T3 in drinking water”
TraceBaseDocs/TraceBaseDocs/Upload/Sample Information Sheet.md:An identifier for the “experiment” or collection of animals that this animal belongs to.
TraceBaseDocs/TraceBaseDocs/Upload/Sample Information Sheet.md:* Matches to the same value in “Samples” sheet.
```
I will change it everywhere.

lparsons commented on TraceBaseDocs/TraceBaseDocs/Upload/How to Upload.md:
```diff
@@ -18,10 +18,15 @@ you initially submit for upload.
+        you must add those non-matching samples to an
+        [LCMS-metadata file](https://docs.google.com/spreadsheets/d/1rfKOGqms8LPeqORO5gyTXLXDU2lvz-CG2aCEwmu8xHw/copy)
+      - If you have a Tissue that is not included in the existing list, please
+        label it as you would like to in the `Tissues` sheet and tell the
+        developer you have a new Tissue.
```
```suggestion
        label it as you would like to in the `Tissues` sheet and inidcate
        that you are submitting a new Tissue Type in the "Special Instructions"
        section of the form.
```

lparsons reviewed (approved):
Having the lc methods spelled out in the form is great. I can't see the LCMS-metadata file template, the link seems broken: "Sorry, unable to open the file at this time. Please check the address and try again.". A few minor comments, but otherwise, this looks good to go now.

hepcat72 commented:
I changed the access of the metadata template to public (like the animal sample template).  You should be able to access it now if you're not logged into your google account - I assume that's how you tested it, since it was restricted, but you have edit access.

hepcat72 commented:
I was thinking before that perhaps having a separate page in the docs that includes an example would help?  I'd have added that yesterday, but was eager to move on...  I know that you'd found it confusing in the design and asked for an example, which I believe seemed to help...

lparsons commented:
Examples can be really useful, but I'd like to keep the instructions as streamlined as simple as possible. Right now we have two major issues to contend with:
1. TraceBase doesn't strore the `mzXML` files
2. Submission of new data is too cumbersome

hepcat72 commented:
I started working on #710 this afternoon. I've been selecting issues to work on based on the issue-tracking issues (#693, #700, and #701).  I don't actually see an issue for storing the mzXML files.  Am I just overlooking it?  If it doesn't exist, I can create an issue for it, if you like.

Also, I have a pretty well-thought-out plan for streamlining the submission process.  I outlined it on pages 2 & 3 of [my original proposal](https://docs.google.com/document/d/1TfpLivNQzDLIB6XJkG1aQwB6SnvuyQuRYqdFZZawYJE/).  Its groundwork is laid out in:

- #671
- #661

But I haven't created issues for the rest of it.  Essentially, all a user would have to do is submit accucor/isocorr files and once errors are addressed, it would spit out an excel file with the usual animals/samples/treatments/tissues tabs plus tabs for study, infusates, tracers, and the LCMS metadata (which in the proposal, is referred to in 2 sheets: corrections and mzxml sheets) with much of the tedious work done automatically.

They wouldn't have to be new tabs in an excel file either.  It could just output tsv files (e.g. compounds, lcms metadata, etc.) like we currently use, though a new tracer/infusate tab would be nice.

But the major benefit would be that it can stub-out all these files with the easy things and the user can just manually fill in the rest (i.e. data that cannot be unambiguously computed).

The one limiting factor is speed.  E.g. They couldn't submit an entire study's accucor files.  But it could work per handful of accucor files and accumulate the results
